xcodeproj/internal/product.bzl

```diff
@@ -4,6 +4,41 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 load(":linker_input_files.bzl", "linker_input_files")
 load(":memory_efficiency.bzl", "EMPTY_DEPSET")
 
+# Values here need to match those in `PBXProductType.swift`
+PRODUCT_TYPE_ENCODED = {
+    "com.apple.product-type.application": "a",
+    "com.apple.product-type.application.messages": "M",
+    "com.apple.product-type.application.on-demand-install-capable": "A",
+    "com.apple.product-type.application.watchapp": "0",
+    "com.apple.product-type.application.watchapp2": "w",
+    "com.apple.product-type.application.watchapp2-container": "c",
+    "com.apple.product-type.app-extension": "e",
+    "com.apple.product-type.app-extension.intents-service": "i",
+    "com.apple.product-type.app-extension.messages": "m",
+    "com.apple.product-type.app-extension.messages-sticker-pack": "s",
+    "com.apple.product-type.tv-app-extension": "t",
+    "com.apple.product-type.extensionkit-extension": "E",
+    "com.apple.product-type.watchkit-extension": "1",
+    "com.apple.product-type.watchkit2-extension": "W",
+    "com.apple.product-type.xcode-extension": "2",
+    "com.apple.product-type.bundle": "B",
+    # Resource bundles are "b"
```

Comment:
Sorry, not sure if I understand how this comment relates to what comes next?

Reply:
It doesn't, it's saying that no other value can use `"b"`, because resource bundles, which are also `com.apple.product-type.bundle`, will use it. I'll move it to be above `com.apple.product-type.bundle`.

Reply:
SGTM!